### PR TITLE
Updating model for training and adding squad_v2 metric for validation

### LINKED ISSUE #539: Add support for Rouge and BLEU to the QuestionAnsweringTask

### LINKED ISSUE #568: Add support for EM and F1 to the QuestionAnsweringTask
## 🚀 Feature

Add F1 and EM metrics

@ethanwharris please add the taskathon tags

### PR DESCRIPTION
**Pull request by gagan3012**
## What does this PR do?
Updating model for training and adding squad_v2 metric for validation

Fixes #539 and Fixes #568 (issue)

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-flash/tree/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-flash/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: flash/text/seq2seq/question_answering/model.py [hunk 1]
```diff
@@ -70,11 +71,14 @@ def __init__(
             rouge_newline_sep=rouge_newline_sep,
             use_stemmer=use_stemmer,
         )
+        self.squad_v2 = load_metric("squad_v2")
```
Comment (tchaton) [inline flash/text/seq2seq/question_answering/model.py:74]: We should `hard-code` anything related to a `dataset` within a model. You could use state to propagate the metric or dataset name to the model directly.
If you need more explication there, please find us on the Gather :)

Comment (ethanwharris) [reply flash/text/seq2seq/question_answering/model.py:74]: That's just a metric though right? Can presumably be used with any data set?

Comment (tchaton) [reply flash/text/seq2seq/question_answering/model.py:74]: I don't think it could. But even then, we shouldn't force users to use the metrics associated to `squad_v2` dataset if they don't want to.

Maybe, let's just to  `metrics: Union[Metric, Callable, Mapping, Sequence, None] = load_metric("squad_v2")`

**Comment by ethanwharris** [comment]
Closing this as it has gone stale. We now use `RougeMetric` from torchmetrics for the question answering task. Still an open issue to enable users to just pass any metric into seq2seq tasks rather than forcing one.
